demo-app/src/main/java/cash/z/ecc/android/sdk/demoapp/demos/getaddress/GetAddressFragment.kt

```diff
@@ -36,33 +40,44 @@ class GetAddressFragment : BaseDemoFragment<FragmentGetAddressBinding>() {
         // have the seed stored
         seed = Mnemonics.MnemonicCode(seedPhrase).toSeed()
 
-        // the derivation tool can be used for generating keys and addresses
+        // converting seed into viewingKey
         viewingKey = runBlocking {
-            DerivationTool.deriveUnifiedViewingKeys(
+            DerivationTool.deriveUnifiedFullViewingKeys(
                 seed,
                 ZcashNetwork.fromResources(requireApplicationContext())
             ).first()
         }
+
+        val network = ZcashNetwork.fromResources(requireApplicationContext())
+        synchronizer = Synchronizer.newBlocking(
+            requireApplicationContext(),
+            network,
+            lightWalletEndpoint = LightWalletEndpoint.defaultForNetwork(network),
+            seed = seed,
+            birthday = network.saplingActivationHeight
+        )
     }
 
     private fun displayAddress() {
-        // a full fledged app would just get the address from the synchronizer
         viewLifecycleOwner.lifecycleScope.launchWhenStarted {
-            val zaddress = DerivationTool.deriveShieldedAddress(
-                seed,
-                ZcashNetwork.fromResources(requireApplicationContext())
-            )
-            val taddress = DerivationTool.deriveTransparentAddress(
-                seed,
-                ZcashNetwork.fromResources(requireApplicationContext())
-            )
-            binding.textInfo.text = "z-addr:\n$zaddress\n\n\nt-addr:\n$taddress"
+            binding.unifiedAddress.apply {
+                val uaddress = synchronizer.getCurrentAddress()
+                text = uaddress
+                setOnClickListener { copyToClipboard(uaddress) }
+            }
+            binding.saplingAddress.apply {
+                val sapling = synchronizer.getLegacySaplingAddress()
+                text = sapling
+                setOnClickListener { copyToClipboard(sapling) }
+            }
+            binding.transparentAddress.apply {
+                val transparent = synchronizer.getLegacyTransparentAddress()
+                text = transparent
+                setOnClickListener { copyToClipboard(transparent) }
+            }
```

Comment:
I'm surprised at this code. I would expect that for the sapling and transparent addresses, the synchronizer would be used to retrieve the unified address, and then separately to extract the sapling an transparent components from that address; ideally, the retrieval of the UA would be done once; this would be robust against future changes such as the introduction of rolling addresses. Having `synchorinizer.getLegacySaplingAddress` and `synchronizer.getLegacyTransparentAddress` as separate methods that don't take the UA as input means that they could potentially return incompatible address components.